packages/core/src/components/region/RegionPopover/RegionPopover.tsx

```diff
@@ -0,0 +1,170 @@
+    buttonActionText: inputButtonActionText,
+  },
+  idkPostalCodeLink: {
+    text: idkPostalCodeLinkText,
+    to: idkPostalCodeLinkTo,
+    icon: { icon: idkPostalCodeLinkIcon, alt: idkPostalCodeLinkIconAlt },
+  },
+  textBeforeLocation = 'Your current location is:',
+  textAfterLocation = 'Use the field below to change it.',
+  description = 'Offers and availability vary by location.',
+  triggerRef,
+  offsetTop = 6,
+  offsetLeft,
+  placement = 'bottom-start',
+}: RegionPopoverProps) {
+  const inputRef = useRef<HTMLInputElement>(null)
+  const [isOpen, setOpen] = useState(true)
+  const { isValidating, ...session } = useSession()
+  const { popover: displayPopover, closePopover } = useUI()
+  const {
+    input,
+    setInput,
+    setLocation,
+    resetInputField,
+    loading,
+    errorMessage,
+    setErrorMessage,
+    isValidationComplete,
+    postalCode,
+  } = useRegion()
+  const { city } = sessionStore.read()
+
+  const location = city ? `${textToTitleCase(city)}, ${postalCode}` : postalCode
+
+  const handleSubmit = async () => {
+    if (!isValidationComplete) {
+      return
+    }
+
+    await setLocation(
+      inputRef.current?.value,
+      inputFieldErrorMessage,
+      session,
+      () => setOpen(false)
+    )
+  }
+
+  const idkPostalCodeLinkProps = {
+    href: idkPostalCodeLinkTo ?? '#',
+    children: (
+      <>
+        {idkPostalCodeLinkText}
+        {!!idkPostalCodeLinkIcon && (
+          <UIIcon
+            name={idkPostalCodeLinkIcon}
+            aria-label={idkPostalCodeLinkIconAlt}
+            width={20}
+            height={20}
+          />
+        )}
+      </>
+    ),
+  }
+
+  const RegionPopoverContent = (
+    <>
+      <span data-fs-region-popover-description>
+        {postalCode ? (
+          <>
+            {textBeforeLocation} <span>{location}</span>. {textAfterLocation}
+          </>
+        ) : (
+          <>{description}</>
+        )}
+      </span>
+      <UIInputField
+        data-fs-region-popover-input
+        id="region-popover-input-postal-code"
+        inputRef={inputRef}
+        label={inputFieldLabel}
+        actionable
+        value={input}
+        onInput={(e) => {
+          errorMessage !== '' && setErrorMessage('')
+          setInput(e.currentTarget.value)
+        }}
+        onSubmit={handleSubmit}
+        onClear={resetInputField}
+        buttonActionText={loading ? '...' : inputButtonActionText}
+        error={errorMessage}
+      />
+      <UILink data-fs-region-popover-link {...idkPostalCodeLinkProps} />
+    </>
+  )
+
+  return (
+    <>
+      {displayPopover.isOpen && (
+        <div className={`${styles.section} section-region-popover`}>
+          <UIPopover
+            data-fs-region-popover
+            title={title}
+            isOpen={isOpen}
+            content={RegionPopoverContent}
+            placement={placement}
+            dismissible
+            onDismiss={() => {
+              setOpen(false)
+              closePopover
```

Comment:
question: Is it necessary? Since there is a `closePopover()` on the Popover's handleDismiss?